File: mandelbrot/mandelbrotset.py
import cmath


class MandelbrotSet:

    def number_belongs_to_set(self, number):
        z = complex(0, 0)
        for _ in range(0, 200):
            z = z ** 2 + number
            if self.bailout(z):
                return False
        return True

    def bailout(self, z):
        return abs(z.real) > 2 or abs(z.imag) > 2

File: src/mandelbrot-set.js
const Complex = require('Complex')

function mandelbrotSet() {

    function numberBelongsToSet(number) {
        let z = new Complex(0,0)

        for (var i = 0; i < 200; i++) {
            z = z.pow(2).add(number);
            if (bailoutReached(z)) return false;
        }

        return true;
    }

    function bailoutReached(z){
        return Math.abs(z.real) > 2 || Math.abs(z.im) > 2
    } 

    return {
        numberBelongsToSet
    } 
}

module.exports = mandelbrotSet();